src/realm/object-store/sync/sync_session.cpp

```diff
@@ -1236,12 +1257,79 @@ void SyncSession::update_configuration(SyncConfig new_config)
         REALM_ASSERT(m_state == State::Inactive || m_state == State::Paused);
         REALM_ASSERT(!m_session);
         REALM_ASSERT(m_config.sync_config->user == new_config.user);
+        // Since this is used for testing purposes only, just update the current sync_config
         m_config.sync_config = std::make_shared<SyncConfig>(std::move(new_config));
         break;
     }
     revive_if_needed();
 }
 
+void SyncSession::update_sync_config_after_migration()
+{
+    // Migration state changed - Update the configuration to
+    // match the new sync mode.
+    bool flx_sync_requested;
+    {
+        util::CheckedLockGuard cfg_lock(m_config_mutex);
+        m_config.sync_config = m_migration_store->convert_sync_config(m_original_sync_config);
+        flx_sync_requested = m_config.sync_config->flx_sync_requested;
+    }
+
+    util::CheckedLockGuard cfg_lock(m_state_mutex);
+    m_needs_subscription_store_updated.emplace(flx_sync_requested);
+}
+
+void SyncSession::update_subscription_store(bool flx_sync_requested)
+{
+    // The session should be closed before updating the FLX subscription store
+    REALM_ASSERT(!m_session);
+
+    // If the subscription store exists and switching to PBS, then clear the store
+    auto& history = static_cast<sync::ClientReplication&>(*m_db->get_replication());
+    if (!flx_sync_requested) {
+        if (m_flx_subscription_store) {
+            history.set_write_validator_factory(nullptr);
+            m_flx_subscription_store.reset();
```

Comment:
do we need to do any clean-up here?

Reply:
There is clean-up for the subscription store, but that will be done in another PR.